static/app/utils/objectFromEntries.ts

```diff
@@ -0,0 +1,25 @@
+type ArrayValue<A> = A extends readonly (infer T)[] ? T : never;
+// as const will cast all properties to readonly
+type RemoveReadOnly<T> = {-readonly [P in keyof T]: RemoveReadOnly<T[P]>};
+
+type FromEntries<T> = T extends [infer Key, any][]
+  ? {[K in Extract<Key, string>]: Extract<ArrayValue<T>, [K, any]>[1]}
+  : {[key in string]: any};
+
+function isIterable<T>(obj: any): obj is Iterable<T> {
+  return typeof obj?.[Symbol.iterator] === 'function';
+}
+
+export function objectFromEntries<T>(entries: T): FromEntries<RemoveReadOnly<T>> {
+  const target = {} as FromEntries<RemoveReadOnly<T>>;
+
+  if (!isIterable<T>(entries)) {
+    throw new TypeError(`Entries is not iterable, got ${JSON.stringify(entries)}`);
```

Comment:
@JonasBa how safe is it to always try to JSON stringify these entries? Any chance this could cause an error?

Reply:
@JonasBa do we really need to stringify the entries then? I'd think Sentry would be able to capture the value of `entries` automatically. I'd also be concerned how the variable error message would impact grouping.

Reply:
Hmm, I dont think it would capture the values unless we throw them right? Also if we just throw entries value, it doesnt accurately reflect the isIterable check here. That's why I opted for the error message + stringifying the value